src/controllers/uploads.js

```diff
@@ -227,11 +227,14 @@ function saveFileToLocal(uploadedFile, callback) {
 			file.saveFileToLocal(filename, 'files', uploadedFile.path, next);
 		},
 		function (upload, next) {
-			next(null, {
+			var storedFile = {
 				url: nconf.get('relative_path') + upload.url,
 				path: upload.path,
 				name: uploadedFile.name,
-			});
+			};
+
+			plugins.fireHook('filter:uploadStored', { uploadedFile: uploadedFile, storedFile: storedFile }, next);
+			next(null, storedFile);
```

Comment:
`plugins.fireHook` calls next itself when it is done, so don't call next a second time.